### PR TITLE
Ensure models don't leak between test through RequestStore, for example User.anonymous

### PR DESCRIPTION
**Pull request by toy**
Stumbled upon [test failure](https://github.com/opf/openproject/actions/runs/17582783029/attempts/1?pr=20243) that got bisected to:
```shell
rspec -fd './modules/storages/spec/requests/api/v3/projects/projects_storages_filter_spec.rb[1:1:1:2:1:2]' './spec/models/principals/scopes/ordered_by_name_spec.rb[1:1:1:1:1,1:1:1:1:2:1,1:1:2:1:1,1:1:2:1:2:1,1:1:3:1:1,1:1:3:1:2:1,1:1:4:1:1,1:1:4:1:2:1,1:1:5:1:1,1:1:5:1:2:1,1:1:6:1:1,1:1:6:1:2:1]' --seed 33618
```
Problem comes from `AnonymousUser` leaking from `after(:all)` by being cached in `RequestStore`.
This ensures that `RequestStore` is empty not only after each test, but also after each context.

### CODE REVIEW

**Review by akabiru — APPROVED** [review]

**Comment by klaustopher** [comment]
Personally, not 100% sure if it makes sense that a side effect of `shared_let` should be that the Request Store is cleared. Wouldn't it make more sense to have that be an `after` callback for every test run?

**Comment by NobodysNightmare** [comment]
I agree to Klaus. I'd also not expect anything to leak between two test executions, so clearing the store after each test seems reasonable.

The current implementation would mean that tests without `shared_let` still can leak through the `RequestStore`, while tests with a shared let can't. That's counter intuitive to me, because `shared_let` usually leads to more (conscious) "leaking".

File: spec/support/rspec_cleanup.rb [hunk 1]
```diff
@@ -16,6 +16,11 @@
     RequestStore.clear!
   end
 
+  config.append_after(:all) do
+    # Ensure models don't leak between test through RequestStore if it is used in after(:all)
+    RequestStore.clear!
```
Comment (NobodysNightmare) [inline spec/support/rspec_cleanup.rb:21]: Now that you put this here: Doesn't line 16 already do exactly that?

Thinking about the `after(:all)`: Shouldn't that be `after(:each)`?

Comment (NobodysNightmare) [reply spec/support/rspec_cleanup.rb:21]: Okay. Nevermind, maybe I understood now. Let me try to explain it and you tell me if I got it:

We already execute the clear after _each_ single example. However, since `after(:all)` callbacks run many times per test suite (at the end of each _group_), and since we have some `after(:all)` callbacks that might touch the `RequestStore`, these could then cause leaking.

About right?

**Review by NobodysNightmare — APPROVED** [review]
LGTM assuming I now understood the edge case we need to fix.

File: spec/support/rspec_cleanup.rb (hunk 1 above)
Comment (toy) [reply spec/support/rspec_cleanup.rb:21]: The problem is exactly in that `after(:each)` is not enough: the [test that brakes it](https://github.com/opf/openproject/blob/dev/modules/storages/spec/requests/api/v3/projects/projects_storages_filter_spec.rb) uses `after(:all)` from which `User.anonymous` is indirectly called leading it to be stored into `RequestStore` after it is cleaned by the `clear!` called from `after` on line 16

Comment (toy) [reply spec/support/rspec_cleanup.rb:21]: > We already execute the clear after _each_ single example. However, since `after(:all)` callbacks run many times per test suite (at the end of each _group_), and since we have some `after(:all)` callbacks that might touch the `RequestStore`, these could then cause leaking.

Exactly that

**Comment by klaustopher** [comment]
Better approach. Good catch.
